Add Possibility to Configure PSK based Encryption on UI

fzoske opened the pull request:
Hi,

This PR add the possibility to configure PSK based encryption for the communication with the Zabbix server (https://www.zabbix.com/documentation/3.0/manual/encryption/using_pre_shared_keys).

Best regards,
Fabian

hudecof commented:
+1

mimugmail commented:
I can have a look next week when I'm back.

mimugmail commented on net-mgmt/zabbix-proxy/src/opnsense/service/templates/OPNsense/Zabbixproxy/zabbix_proxy.conf:
```diff
@@ -61,4 +61,13 @@ Timeout={{ OPNsense.zabbixproxy.general.timeout }}
+{% else %}
+TLSConnect=psk
+{% endif %}
+TLSPSKFile=/usr/local/etc/zabbix3/zabbix_proxy/zabbix_proxy.psk
```
The directory is wrong, you should use the same from targets file

mimugmail commented on net-mgmt/zabbix-proxy/src/opnsense/mvc/app/models/OPNsense/Zabbixproxy/General.xml:
```diff
@@ -79,5 +79,17 @@
+        </encryptionid>
+        <encryptionkey type="TextField">
+            <default></default>
+            <Required>N</Required>
```
Can you add the correct validations to ID and PSK please? https://www.zabbix.com/documentation/3.0/manual/encryption/using_pre_shared_keys

fzoske commented:
@mimugmail Fix it and renamed the config ids

One suggestion: Split the general and the encryption config to two tabs and namespaces (like the config of the Zabbix Agent plugin).
e.g.: `OPNsense.zabbixproxy.general.encryptionpsk` => `OPNsense.zabbixproxy.encryption.psk`

What do you think about this?

mimugmail commented:
Hm, I have no opinion on this. I'm fine with General, hide with Advanced or a new tab.

fzoske commented:
@mimugmail I put them under advanced for now

mimugmail commented on net-mgmt/zabbix-proxy/src/opnsense/mvc/app/models/OPNsense/Zabbixproxy/General.xml:
```diff
@@ -79,5 +79,24 @@
+            <mask>/^.*{1,128}$/</mask>
+            <ValidationMessage>Should be a string between 1 and 128 characters.</ValidationMessage>
+            <advanced>true</advanced>
+        </encryptionid>
```
Wrong closing here ...

fzoske commented:
fixed

fzoske commented:
@fichtner You're right, the quantifier was wrong there. Fixed it.

mimugmail commented:
Ok, now it's working fine. 
One thing, on my side it's not under advanced .. @fichtner does this requires a model bump?

@fzoske can you also bump the version in Makefile?

fichtner commented:
yes to model version bump

fichtner commented on net-mgmt/zabbix-proxy/src/opnsense/mvc/app/models/OPNsense/Zabbixproxy/General.xml:
```diff
@@ -79,5 +79,24 @@
+        <encryption type="BooleanField">
+            <default>0</default>
+            <Required>Y</Required>
+            <advanced>true</advanced>
```
@fzoske oh, @mimugmail is right. advanced is for the form, not the model

last bit before merge... many thanks so far!

fzoske commented:
Alright. Moved it to the form.... Somehow I missed that

fichtner commented:
Merged, thanks!
